federation/pkg/federation-controller/util/cluster_util.go

```diff
@@ -18,21 +18,25 @@ package util
 
 import (
 	"fmt"
+	"net"
+	"os"
+	"time"
+
 	"github.com/golang/glog"
 	federation_v1alpha1 "k8s.io/kubernetes/federation/apis/federation/v1alpha1"
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/client/restclient"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
 	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
 	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
 	utilnet "k8s.io/kubernetes/pkg/util/net"
-	"net"
-	"os"
 )
 
 const (
 	KubeAPIQPS              = 20.0
 	KubeAPIBurst            = 30
 	KubeconfigSecretDataKey = "kubeconfig"
+	getSecretTimeout        = 1 * time.Minute
```

Comment:
Did you mean: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/util.go?
Thats the package that has standard timeouts.
But I cannot depend on test code from here. We can perhaps refactor that library to move generic parts out of test.